src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectProtocolValidator.cs

```diff
@@ -403,5 +550,34 @@ protected virtual void ValidateNonce(JwtSecurityToken jwt, OpenIdConnectProtocol
                 }
             }
         }
+
+        /// <summary>
+        /// Validates that the 'state' in message is valid.
+        /// </summary>
+        /// <param name="validationContext">a <see cref="OpenIdConnectProtocolValidationContext"/> that contains the 'state' to validate.</param>
+        /// <exception cref="ArgumentNullException">if 'validationContext' is null.</exception>
+        /// <exception cref="OpenIdConnectProtocolInvalidStateException">if 'state' in the context does not match the state in the message.</exception>
+        protected virtual void ValidateState(OpenIdConnectProtocolValidationContext validationContext)
+        {
+            if (!RequireState)
+            {
+                return;
+            }
+
+            if (validationContext == null)
+            {
+                LogHelper.Throw(string.Format(CultureInfo.InvariantCulture, ErrorMessages.IDX10000, GetType() + ": validationContext"), typeof(ArgumentNullException), EventLevel.Verbose);
+            }
+
+            if (string.IsNullOrEmpty(validationContext.State) || string.IsNullOrEmpty(validationContext.ProtocolMessage.State))
+            {
+                LogHelper.Throw(ErrorMessages.IDX10327, typeof(OpenIdConnectProtocolInvalidStateException), EventLevel.Error);
+            }
+
+            if (!string.Equals(validationContext.State, validationContext.ProtocolMessage.State, StringComparison.Ordinal))
+            {
+                LogHelper.Throw(string.Format(CultureInfo.InvariantCulture, ErrorMessages.IDX10328, validationContext.State ?? "null", validationContext.ProtocolMessage.State ?? "null"), typeof(OpenIdConnectProtocolInvalidStateException), EventLevel.Error);
```

Comment:
I am referring to the checks in the log message: validationContext.State ?? "null", validationContext.ProtocolMessage.State ?? "null") as they wont be null at this point because of the if statement right above: if(string.IsNullOrEmpty(validationContext.State) || string.IsNullOrEmpty(validationContext.ProtocolMessage.State))